receiver/receivercreator/runner.go

```diff
@@ -110,5 +111,7 @@ func (run *receiverRunner) createRuntimeReceiver(
 	cfg config.Receiver,
 	nextConsumer consumer.Metrics,
 ) (component.MetricsReceiver, error) {
-	return factory.CreateMetricsReceiver(context.Background(), run.params, cfg, nextConsumer)
+	runParams := run.params
+	runParams.Logger = runParams.Logger.With(zap.String("name", cfg.ID().String()))
```

Comment:
It may still be useful to now the receiver comes from the receiver creator (and to know from which receiver creator it comes from, if there are several), so maybe we should use a different key here. What do you think?

Reply:
Definitely useful, and I think this information is already provided in the receiver's generated config id name: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d0c76dec709dcac5762c4313a536cb2f3d724264/receiver/receivercreator/runner.go#L103

The runner's `idNamespace` is the parent receiver creator's config id.

Reply:
Thanks, I wasn't aware of that :)